subsys/net/lib/sockets/Kconfig

```diff
@@ -96,6 +96,18 @@ config NET_SOCKETS_OFFLOAD
 	  See NET_OFFLOAD for a more deeply integrated approach which offloads
 	  from the net_context() API within the Zephyr IP stack.
 
+config NET_SOCKETS_PACKET
+	bool "Enable packet socket support"
+	default n
+	help
+	  This is an initial version of packet socket support (special type
+	  raw socket). Packets are passed to and from the device driver
+	  without any changes in the packet headers. It's API caller
+	  responsibility to provide all the headers (e.g L2, L3 and so on)
+	  while sending. While receiving, packets (including all the headers)
+	  will be feed to sockets as it as from the driver. This version
+	  supports only over Etherent.
```

Comment:
Last two sentences are cryptic, could we say here:

```
When packets are received, the incoming data which includes all the protocol headers are feed to the listening socket.
This version supports only communication over Ethernet.
```

Because we only support Ethernet atm, please add  dependencies properly. So "depends on NET_L2_ETHERNET"